libsolidity/ast/Types.cpp

```diff
@@ -1465,6 +1534,11 @@ bool StructType::isImplicitlyConvertibleTo(const Type& _convertTo) const
 	return this->m_struct == convertTo.m_struct;
 }
 
+string StructType::identifier() const
+{
+	return "t_struct_" + m_struct.name() + "_" + std::to_string(m_struct.id()) + identifierLocationSuffix();
```

Comment:
Here `m_struct.name()` can be `23_memory_array`.  The whole identifier becomes `t_struct_S_23_memory_array_33_memory`.